### PR TITLE
[MOB-4781] userdefaults migration refactor

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 Jira Ticket(s)

* [MOB-4781](https://iterable.atlassian.net/browse/MOB-4781)

## ✏️ Description

* creates a new migration function in `LocalStorage` for `lastPushPayload`
* cleans up/fixes the logic of the auth data migration a little bit for unexpected edge cases

### CODE REVIEW

File: swift-sdk/Internal/IterableUserDefaults.swift [hunk 1]
```diff
@@ -26,6 +28,7 @@ class IterableUserDefaults {
         }
     }
     
+    // migrated to IterableKeychain
     var authToken: String? {
```
[the remark this answers is not in the document]
Comment (roninopf) [reply swift-sdk/Internal/IterableUserDefaults.swift:32]: Yes, they need to stay there for the migration function to work.

File: swift-sdk/Internal/LocalStorage.swift [hunk 2]
```diff
@@ -99,35 +99,45 @@ struct LocalStorage: LocalStorageProtocol {
...
+        
+        if let userDefaultAuthToken = userDefaultAuthToken {
+            if keychain.authToken == nil {
+                keychain.authToken = userDefaultAuthToken
+            }
+            
+            iterableUserDefaults.authToken = nil
+            
+            ITBInfo("UPDATED: migrated authToken from UserDefaults to IterableKeychain")
         }
```
[the remark this answers is not in the document]
Comment (roninopf) [reply swift-sdk/Internal/LocalStorage.swift:132]: That's correct that it's already stored. And in that case, we should only rely on what's in the keychain and not migrate.

[the remark this answers is not in the document]
Comment (evangreer91) [reply swift-sdk/Internal/LocalStorage.swift:132]: Got it

File: swift-sdk/Internal/LocalStorage.swift [hunk 3]
```diff
@@ -99,35 +99,45 @@ struct LocalStorage: LocalStorageProtocol {
...
     }
     
     private func moveLastPushPayloadFromUserDefaultsToKeychain() {
-        if let (userDefaultLastPushPayload, expiration) = iterableUserDefaults.getLastPushPayloadAndExpirationPair() {
+        if let (userDefaultLastPushPayload, expiration) = iterableUserDefaults.getLastPushPayloadExpirationPairForMigration() {
             keychain.setLastPushPayload(userDefaultLastPushPayload, withExpiration: expiration)
+            iterableUserDefaults.save(payload: nil, withExpiration: nil)
             
-            ITBInfo("UPDATED: moved lastPushPayload from UserDefaults to IterableKeychain")
+            ITBInfo("UPDATED: migrated lastPushPayload from UserDefaults to IterableKeychain")
```
[the remark this answers is not in the document]
Comment (evangreer91) [reply swift-sdk/Internal/LocalStorage.swift:140]: Ah got it

**Review by evangreer91 — APPROVED** [review]

**Pull request merged**